command.go

```diff
@@ -1147,6 +1175,10 @@ func (c *Command) AddCommand(cmds ...*Command) {
 			panic("Command can't be a child of itself")
 		}
 		cmds[i].parent = c
+		// if Group is not defined generate a new one with same title
+		if x.Group != "" && !c.ContainsGroup(x.Group) {
+			c.AddGroup(&Group{Group: x.Group, Title: x.Group})
+		}
```

Comment:
I wonder what is better for the user?  
Implicitly creating the group seems nice at first impression.  However, I wonder if it makes typos harder to notice?
Say I want a group of commands called "System:" (notice the colon), and for one of my commands I type `Group: "System"` with a missing colon.  This will implicitly create a new group.  It will be obvious looking at the help, but if this is a couple of levels of subcommands down, it may go unnoticed.

What we could do instead is `panic` if the group was not created in advance.  That way, the user will not miss such subtle errors.

The flip side of course is that if forces users to call `AddGroup` for each group.  Is that too heavy?

A benefit is that it forces users to specify a `Title` for the group, instead of sometimes using the id (`Group.Group`) as the title.

And a nice side-effect is that it would make the order of groups more obvious: groups will be displayed in the order they were created with `AddGroup()`

Any opinions on that?

Reply:
> What we could do instead is panic if the group was not created in advance. That way, the user will not miss such subtle errors.

I like the idea of not implicitly creating groups now that you bring it up but is panicing the right move here? Do we have a panic president elsewhere in cobra to surface user error?

> And a nice side-effect is that it would make the order of groups more obvious: groups will be displayed in the order they were created with AddGroup()

Yes, I think that's the obvious benefit. The unknown ordering of groups in the help messaging would be a poor user experience for people creating CLIs. My vote would be to be more explicit.

Reply:
> I like the idea of not implicitly creating groups now that you bring it up but is panicing the right move here? 
> Do we have a panic president elsewhere in cobra to surface user error?

Two lines above 😉 
https://github.com/spf13/cobra/blob/212ea4078323771dc49b6f25a41d84efbaac3a4c/command.go#L1177-L1179
I rarely have original thoughts 🤣 

I have to admit that using `panic()` is something I used to think was exaggerated.   However, I now feel that as long as we know that the panic would be triggered immediately, then we know a developer would immediately realize their error and correct it.

Reply:
I'll change that such that it panics. I agree that implicitly defining groups was maybe a bad idea.

Reply:
is changed.